hnn_core/network.py

```diff
@@ -346,13 +368,14 @@ class Network(object):
 
     Notes
     ----
-    `net = default_network(params)` is the reccomended path for creating a
+    `net = default_network(params)` is the recommended path for creating a
     network. Instantiating the network as `net = Network(params)` will
-    produce a network with no cell to cell connections. As such,
+    produce a network with no cell-to-cell connections. As such,
     connectivity information contained in `params` will be ignored.
     """
 
     def __init__(self, params, add_drives_from_params=False,
+                 inplane_distance=1., layer_separation=1307.4,
                  legacy_mode=True):
```

Comment:
What if instead of creating an `inplane_distance` argument, we create a `cell_grid` argument that also exposes the number of cell replications in the network? The idea here would be to limit the arbitrary hard-coded assumptions (e.g., why are there only 100 L5 pyramidal cells in a 10-by-10 grid?) in a principled manner. For example, the `cell_grid=` argument might look like:

```python
grid_cell = {'L2_basket': [5, 7, 1],
             'L2_pyramidal': [10, 10, 1],
             'L5_basket': [10, 10, 1],
             'L5_pyramidal': [10, 10, 1]}
```
where the list assigned to each cell type has the form `[n_xdim, n_ydim, inplane_distance]`.

If this feels like over-engineering for this PR, we could reduce the `cell_grid` argument to a 3-element tuple (same structure as above) that only applies to pyramidal cells.

Reply:
There's still the matter of how the basket cells are distributed (as noted by @jasmainak above). I'm not disagreeing with the objective to remove hard-coded assumptions, but when we do, we'll need quite a complicated API to support arbitrary arrangements. I'd say leave it for a later PR?